src/libmongoc/tests/unified/entity-map.c

```diff
@@ -948,6 +948,10 @@ entity_client_new(entity_map_t *em, bson_t *bson, bson_error_t *error)
       }
       bson_free(test_username);
       bson_free(test_password);
+   } else {
+      // "mongodb+srv://" URIs may apply an authSource=admin default in the TXT record. Override with $external if
+      // testing OIDC.
+      mongoc_uri_set_auth_source(uri, "$external");
```

Comment:
I am confused. Is this not already being handled by mongoc_uri_get_auth_source? Is this stating that the SRV TXT records incorrectly include an authSource=admin when it should be $external for OIDC?